src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -117,81 +241,192 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
-                ->setAllowedTypes(['array'])
-                ->setAllowedValues([static function ($value) {
-                    foreach ($value as $functionName) {
-                        if (!\is_string($functionName) || '' === \trim($functionName) || \trim($functionName) !== $functionName) {
-                            throw new InvalidOptionsException(\sprintf(
-                                'Each element must be a non-empty, trimmed string, got "%s" instead.',
-                                \is_object($functionName) ? \get_class($functionName) : \gettype($functionName)
-                            ));
-                        }
-                    }
+        $exclude = $this->normalizeFunctionNames($this->configuration['exclude']);
 
-                    return true;
-                }])
-                ->setDefault([])
-                ->getOption(),
-        ]);
+        if (in_array(self::SET_ALL, $this->configuration['include'], true)) {
+            if (count($exclude) > 0) {
+                return static function ($functionName) use ($exclude) {
+                    return !isset($exclude[strtolower($functionName)]);
+                };
+            }
+
+            return static function () {
+                return true;
+            };
+        }
+
+        $include = [];
+        if (in_array(self::SET_INTERNAL, $this->configuration['include'], true)) {
+            $include = $this->getAllInternalFunctionsNormalized();
+        } elseif (in_array(self::SET_COMPILER_OPTIMIZED, $this->configuration['include'], true)) {
+            $include = $this->getAllCompilerOptFunctionsNormalized(); // if `@internal` is set all compiler opt function are already loaded
+        }
+
+        foreach ($this->configuration['include'] as $additional) {
+            if ('@' !== $additional[0]) {
+                $include[strtolower($additional)] = true;
+            }
+        }
+
+        if (count($exclude) > 0) {
+            return static function ($functionName) use ($include, $exclude) {
+                return isset($include[strtolower($functionName)]) && !isset($exclude[strtolower($functionName)]);
+            };
+        }
+
+        return static function ($functionName) use ($include) {
+            return isset($include[strtolower($functionName)]);
+        };
     }
 
     /**
-     * @return string[]
+     * @return array<string, true> normalized function names of which the PHP compiler optimizes
      */
-    private function getFunctionNames()
+    private function getAllCompilerOptFunctionsNormalized()
     {
-        $definedFunctions = \get_defined_functions();
+        static $optFunctions = null;
 
-        return \array_diff(
-            $this->normalizeFunctionNames($definedFunctions['internal']),
-            \array_unique($this->normalizeFunctionNames($this->configuration['exclude']))
-        );
+        if (null === $optFunctions) {
+            $optFunctions = $this->normalizeFunctionNames([
```

Comment:
Re-checked a list on php-src - all good. Also 'opt' is here instead of 'optimized' as mentioned earlier.

Reply:
renamed all the `opt` to `optimized`